ext/nnstreamer/tensor_filter/tensor_filter_nnfw.c

```diff
@@ -104,6 +106,62 @@ nnfw_get_accelerator (nnfw_pdata * pdata, const char *accelerators)
   }
 }
 
+/**
+ * @brief Verify the provided model file matches the existing metadata file
+ */
+static int
+nnfw_metadata_verify (const char *meta_file, const char *model_file)
+{
+  int err = -EINVAL;
+  JsonParser *parser;
+  JsonNode *root;
+  JsonNode *model_node;
+  GError *error;
+  JsonObject *object;
+  JsonArray *filenames;
+  GValue model_value = G_VALUE_INIT;
+  gchar *basename_ref, *basename_in;
+  const gchar *filename;
+
+  parser = json_parser_new ();
+  if (json_parser_load_from_file (parser, meta_file, &error)) {
+    root = json_parser_get_root (parser);
+    object = json_node_get_object (root);
+
+    /** Parse the list of models from the metadata file */
+    if (object && json_object_has_member (object, "models")) {
+      filenames = json_object_get_array_member(object, "models");
+
+      /** Ensure that the first element is the model file to be run */
+      if (filenames && json_array_get_length (filenames) > 0) {
+        model_node = json_array_get_element (filenames, 0);
+        json_node_get_value (model_node, &model_value);
+
+        if (G_VALUE_HOLDS_STRING (&model_value)) {
+          filename = g_value_get_string (&model_value);
+
+          basename_ref = g_path_get_basename (model_file);
+          basename_in = g_path_get_basename (filename);
+          if (g_strcmp0 (basename_in, basename_ref) == 0) {
+            err = 0;
+          }
+
+          g_free (basename_ref);
+          g_free (basename_in);
+        }
+        g_value_unset (&model_value);
+      }
+    }
+  } else {
+    err = error->code;
```

Comment:
Q: Is this a negative or positive value?

Reply:
good catch, its positive. Updated.